tests/test_parser.py

```diff
@@ -748,3 +748,15 @@ def test_label_invalid(self, dfparser, value):
         ])
         with pytest.raises(Exception):
             dfparser.labels
+
+    def test_remove_whitespace(self, tmpdir):
+        """
+        Verify keys are parsed correctly even if there is no final newline.
+
+        """
+        with open(os.path.join(str(tmpdir), 'Dockerfile'), 'wb') as fp:
+            fp.write('FROM scratch')
+        tmpdir_path = str(tmpdir.realpath())
+        df1 = DockerfileParser(tmpdir_path)
+        df1.labels['foo'] = 'bar'
+        assert 'scratchLABEL' not in df1.content
```

Comment:
Might be better to assert what we do expect, e.g. that if we make another DockerfileParser, a df2.from == scratch and df2.labels is as expected.

Reply:
Fixed.